File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/interactive/MapObstacle.scala





















// Generated on 05/03/2013 10:35:07
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait MapObstacle extends TypeDeserializer {
  val typeId = 200

  protected trait Type extends BaseType with Serializable {
    val typeId = MapObstacle.typeId

    val obstacleCellId: Short
    val state: Byte

    def serialize(buf: Buffer) {
      buf.writeShort(obstacleCellId)
      buf.writeByte(state)
    }

  }

  protected def create(obstacleCellId: Short, state: Byte): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val obstacleCellId = buf.readShort()
    val state = buf.readByte()

    create(obstacleCellId, state)
  }

}

object MapObstacle extends MapObstacle {
  case class TypeImpl(obstacleCellId: Short, state: Byte) extends Type
  override def create(obstacleCellId: Short, state: Byte) = TypeImpl(obstacleCellId, state)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(obstacleCellId: Short, state: Byte) = create(obstacleCellId, state)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/MapObstacle.java


// Generated on 05/08/2013 19:38:02
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class MapObstacle extends NetworkType {
    public static final short TYPE_ID = 200;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public short obstacleCellId;
    public byte state;
    
    public MapObstacle() { }
    
    public MapObstacle(short obstacleCellId, byte state) {
        this.obstacleCellId = obstacleCellId;
        this.state = state;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(obstacleCellId);
        buf.writeByte(state);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        obstacleCellId = buf.readShort();
        if (obstacleCellId < 0 || obstacleCellId > 559)
            throw new RuntimeException("Forbidden value on obstacleCellId = " + obstacleCellId + ", it doesn't respect the following condition : obstacleCellId < 0 || obstacleCellId > 559");
        state = buf.readByte();
        if (state < 0)
            throw new RuntimeException("Forbidden value on state = " + state + ", it doesn't respect the following condition : state < 0");
    }
    
}
